forte/data/data_pack.py

```diff
@@ -546,13 +555,19 @@ def set_text(
         text_payload_index: int = 0,
     ):
         """
-        Set text for TextPayload at a specified index.
+        Set text for TextPayload at a specified index or add a new TextPayload
+        in the DataPack.
+
+        Raises:
+            ValueError: raised when the text payload index is out of range.
 
         Args:
             text: a str text.
             replace_func: function that replace text. Defaults to None.
             text_payload_index: the zero-based index of the TextPayload
-                in this DataPack's TextPayload entries. Defaults to 0.
+                in this DataPack's TextPayload entries.
+                If it's -1, a new TextPayload will be added to the DataPack.
+                If it's 0, it adds a new TextPayload if there is no text payload in the data pack.
```

Comment:
The explanation for the `0` part is a bit hard to understand

Reply:
It's a bit tricky indeed because I designed so for backward compatibility. Past `set_text` set/add the the first text payload depending on the condition of the `DataPack.text_payloads`.
* if `DataPack.text_payloads` is empty, it adds a text payload
* if `DataPack.text_payloads` is not empty, it reset the data in the first text payload

What I wanted to is only reset the first text payload and throw error when `DataPack.text_payloads`  is empty. Maybe I can create another issue for it and it could be fixed later together with all test cases modifications.

Reply:
If we throw an error when `text_payloads` is empty, does it mean that some previous usage will break?

Let's probably keep backward compatibility here.

Reply:
Yes throwing error will break the past test cases. The current implementation ensures backward compatibility.